source/common/grpc/google_async_client_impl.cc

```diff
@@ -167,7 +167,24 @@ GoogleAsyncStreamImpl::GoogleAsyncStreamImpl(GoogleAsyncClientImpl& parent,
       service_full_name_(service_full_name), method_name_(method_name), callbacks_(callbacks),
       options_(options), unused_stream_info_(Http::Protocol::Http2, dispatcher_.timeSource(),
                                              Network::ConnectionInfoProviderSharedPtr{},
-                                             StreamInfo::FilterState::LifeSpan::FilterChain) {}
+                                             StreamInfo::FilterState::LifeSpan::FilterChain) {
+  if (nullptr != options.parent_span_) {
```

Comment:
Nit: prefer the other way around in Envoy style. I know why this is done historically for C-derived languages, but it's quite jarring when the rest of the code base doesn't follow this style and modern compilers are fairly good at picking up mistakes here.
```suggestion
  if (options.parent_span_ != nullptr) {
```

Reply:
TIL about Yoda conditions, thank you. I'm new to the language :)